aws-ssm-document/src/main/java/com/amazonaws/ssm/document/DocumentModelTranslator.java

```diff
@@ -164,10 +170,22 @@ private String generateName(final Map<String, String> systemTags, final String r
         return Optional.of(stackName);
     }
 
-    private String processDocumentContent(final Object content) {
+    private String processDocumentContent(final Object content, final String documentFormat) {
         if (content instanceof Map) {
+            final ObjectMapper mapper;
+
+            // DocumentFormat is not required, default DocumentFormat is JSON.
+            if (documentFormat == null) {
+                mapper = mappers.get(DocumentFormat.JSON.toString());
+            } else {
+                mapper = mappers.get(documentFormat);
+            }
```

Comment:
nit: format = Optional.of(documentFormat).orElse(DocumentFormat.JSON.toString());